server/src/handlers/file_handler.rs

```diff
@@ -298,10 +299,22 @@ pub struct FileData {
 pub async fn get_dataset_files_handler(
     data: web::Path<DatasetFileQuery>,
     pool: web::Data<Pool>,
-    _dataset_org_plan_sub: DatasetAndOrgWithSubAndPlan,
-    _required_user: LoggedUser,
+    dataset_org_plan_sub: DatasetAndOrgWithSubAndPlan,
+    required_user: LoggedUser,
 ) -> Result<HttpResponse, actix_web::Error> {
     let data = data.into_inner();
+    if dataset_org_plan_sub.organization.organization.id != data.dataset_id.clone() {
+        return Err(ServiceError::BadRequest(
+            "Organization ID does not match dataset ID".to_string(),
+        )
+        .into());
+    }
```

Comment:
```suggestion
    if dataset_org_plan_sub.organization.organization.id != data.organization_id.clone() {
        return Err(ServiceError::BadRequest(
            "Organization ID does not match dataset ID".to_string(),
        )
        .into());
    }
```

Seems like this would be wrong.